File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
import java.util.HashMap;
// import java.lang.Math.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        // HashMap
        int longest = 0;
        HashMap<Character, Integer> got = new HashMap<Character, Integer>();
        for (int i = 0; i < s.length(); i++) {
            if (got.containsKey(s.charAt(i))) {
                longest = Math.max(longest, got.size());
                i = ogt.get(s.charAt(i));
                got.clear();
            } else {
                got.put(s.charAt(i), i);
            }
        }

        return Math.max(longest, got.size());
    }
}t-repeating-characterslongest-substring-without-repeating-characters

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longest = 0
        length = len(s)
        got = {}
        i = 0
        c = 's'
        while(i<length):
            c = s[i]
            if c not in got:
                got[c] = i
                i+=1
            else :
                longest = max(len(got),longest);
                i = got[c]+1
                got.clear()

        return max(len(got),longest)
